eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelayer.scala

```diff
@@ -38,33 +38,42 @@ object NodeRelayer {
   // @formatter:off
   sealed trait Command
   case class Relay(nodeRelayPacket: IncomingPacket.NodeRelayPacket) extends Command
+  case class RelayComplete(childHandler: ActorRef[NodeRelay.Command], paymentHash: ByteVector32) extends Command
+  private[relay] case class GetPendingPayments(replyTo: akka.actor.ActorRef) extends Command
   // @formatter:on
 
   def mdc: Command => Map[String, String] = {
-    case c: Relay => Logs.mdc(
-      paymentHash_opt = Some(c.nodeRelayPacket.add.paymentHash))
+    case c: Relay => Logs.mdc(paymentHash_opt = Some(c.nodeRelayPacket.add.paymentHash))
+    case c: RelayComplete => Logs.mdc(paymentHash_opt = Some(c.paymentHash))
+    case _: GetPendingPayments => Logs.mdc()
   }
 
-  def apply(nodeParams: NodeParams, router: ActorRef, register: ActorRef): Behavior[Command] =
+  def apply(nodeParams: NodeParams, router: akka.actor.ActorRef, register: akka.actor.ActorRef, children: Map[ByteVector32, ActorRef[NodeRelay.Command]] = Map.empty): Behavior[Command] =
```

Comment:
You scraped that comment:
> // we index children by paymentHash, not relayId, because there is no concept of individual payment on LN

It is arguably a bit cryptic but maybe this one is better?
```suggestion
    /**
   * @param children a map of current in-process payments, indexed by payment hash and purposefully *not* by payment id,
   *                 because that is how we aggregate payment parts.
   */
  def apply(nodeParams: NodeParams, router: akka.actor.ActorRef, register: akka.actor.ActorRef, children: Map[ByteVector32, ActorRef[NodeRelay.Command]] = Map.empty): Behavior[Command] =
```

Reply:
That previous comment was really confusing, your proposal is much better!

Reply:
Done in https://github.com/ACINQ/eclair/pull/1716/commits/0ddbc2b02f8988c3c8f6634432d7c917193b037e